Strange warning gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4:

src/library/columncache.cpp: In member function ‘void ColumnCache::slotSetKeySortOrder(double)’:
src/library/columncache.cpp:97:6: warning: parameter ‘notation’ set but not used [-Wunused-but-set-parameter]

```    if (notation != static_cast<double>(KeyUtils::LANCELOT)) {```

helps.

--- a/src/library/columncache.cpp
+++ b/src/library/columncache.cpp
@@ -187,7 +187,7 @@ void ColumnCache::slotSetKeySortOrder(double notationValue) {
     QString keySortSQL("CASE %1_id WHEN NULL THEN 0 ");
     for (int i = 0; i <= 24; ++i) {
             keySortSQL.append(QString("WHEN %1 THEN %2 ")
-                .arg(sortOrder[i]).arg(i));
+                .arg(QString::number(sortOrder[i]), QString::number(i)));
     }
     keySortSQL.append("END");
 

--- a/src/library/columncache.cpp
+++ b/src/library/columncache.cpp
@@ -94,12 +94,9 @@ void ColumnCache::setColumns(const QStringList& columns) {
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_PLAYLISTTRACKSTABLE_TITLE], sortNoCase);
 }
 
-void ColumnCache::slotSetKeySortOrder(double notationValue) {
-    // MSVC++ 2013 requires casting to int before enum
-    KeyUtils::KeyNotation notation = static_cast<KeyUtils::KeyNotation>(
-                                       static_cast<int>(notationValue));
+void ColumnCache::slotSetKeySortOrder(double notation) {
     std::vector<mixxx::track::io::key::ChromaticKey> sortOrder;
-    if (notation != KeyUtils::LANCELOT) {
+    if (notation != static_cast<double>(KeyUtils::LANCELOT)) {
         sortOrder = {
             mixxx::track::io::key::INVALID,
 

--- a/src/library/columncache.cpp
+++ b/src/library/columncache.cpp
@@ -94,9 +94,12 @@ void ColumnCache::setColumns(const QStringList& columns) {
     m_columnSortByIndex.insert(m_columnIndexByEnum[COLUMN_PLAYLISTTRACKSTABLE_TITLE], sortNoCase);
 }
 
-void ColumnCache::slotSetKeySortOrder(double notation) {
+void ColumnCache::slotSetKeySortOrder(double notationValue) {
+    // MSVC++ 2013 requires casting to int before enum
+    KeyUtils::KeyNotation notation = static_cast<KeyUtils::KeyNotation>(
+                                       static_cast<int>(notationValue));
     std::vector<mixxx::track::io::key::ChromaticKey> sortOrder;
-    if (static_cast<KeyUtils::KeyNotation>(notation) != KeyUtils::LANCELOT) {
+    if (notation != KeyUtils::LANCELOT) {
         sortOrder = {
             mixxx::track::io::key::INVALID,
 

